flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -137,7 +137,7 @@ extends Actor with ActorLogMessages with ActorLogging {
   protected val resources = HardwareDescription.extractFromSystem(memoryManager.getMemorySize)
 
   /** Registry of all tasks currently executed by this TaskManager */
-  protected val runningTasks = scala.collection.mutable.HashMap[ExecutionAttemptID, Task]()
+  protected val runningTasks = scala.collection.concurrent.TrieMap[ExecutionAttemptID, Task]()
```

Comment:
I have recently read (I think in the Databricks Scala guide) that they discourage the Scala concurrent package, because of bugs. How about using the java concurrent HashMap? That one has pretty good performance seems to work reliably.